### PR TITLE
Max characters

### PR DESCRIPTION
**Pull request by thnom**
Additional parameter for TextInput to limit the total number of characters input

### CODE REVIEW

File: Source/OctaneGUI/Controls/TextInput.cpp [hunk 1]
```diff
@@ -890,6 +891,11 @@ void TextInput::AddText(const std::u32string& Contents)
     {
         return;
     }
+    
+    if (m_Text->Length() + Contents.length() > m_MaxCharacters)
```
Comment (mdavisprog) [inline Source/OctaneGUI/Controls/TextInput.cpp]: This should check to see if m_MaxCharacters > 0. If it is zero, the control should have no limit.

File: Source/OctaneGUI/Controls/TextInput.cpp [hunk 2]
```diff
@@ -890,6 +891,11 @@ void TextInput::AddText(const std::u32string& Contents)
...
+    {
+        return;
+    }
```
Comment (mdavisprog) [inline Source/OctaneGUI/Controls/TextInput.cpp]: The contents of the incoming string should be stripped to allow the control to meet the m_MaxCharacters limit e.g. the m_MaxCharacters = 2 and the incoming string is "Hello", the control should have the text "He".

File: Source/OctaneGUI/Controls/TextInput.h [hunk 3]
```diff
@@ -181,6 +181,7 @@ class TextInput : public ScrollableViewControl
     bool m_Multiline { false };
     bool m_ReadOnly { false };
     bool m_NumbersOnly { false };
+    uint32_t m_MaxCharacters { 0 };
```
Comment (mdavisprog) [inline Source/OctaneGUI/Controls/TextInput.h:187]: Should add a SetMaxCharacters/MaxCharacters set/get function similar to SetNumbersOnly/NumbersOnly.

File: Source/OctaneGUI/Controls/TextInput.cpp (hunk 1 above)
Comment (thnom) [reply Source/OctaneGUI/Controls/TextInput.cpp]: Resolved in https://github.com/thnom/OctaneGUI/commit/079fb6b516a88c2a4334a7b7518609cf522d5bb9

Taken approach with if to do an _and_ but let me know if you would prefer it to check == 0 as its own block

File: Source/OctaneGUI/Controls/TextInput.h (hunk 3 above)
Comment (thnom) [reply Source/OctaneGUI/Controls/TextInput.h:187]: Added in https://github.com/thnom/OctaneGUI/commit/079fb6b516a88c2a4334a7b7518609cf522d5bb9

File: Source/OctaneGUI/Controls/TextInput.cpp (hunk 2 above)
Comment (thnom) [reply Source/OctaneGUI/Controls/TextInput.cpp]: Moved within https://github.com/thnom/OctaneGUI/commit/079fb6b516a88c2a4334a7b7518609cf522d5bb9 to use _Stripped_ version of.

**Comment by thnom** [comment]
Couple of things:
I've changed `size` to `length` as part of https://github.com/mdavisprog/OctaneGUI/pull/3/commits/f72b3f04c4eb88a878dbbd42a5f283859f2bb9d9. Think this change _is_ the right behaviour as leaving as-is would return a stupidly high number rather than say 5 for "hello"

The overview example will _still_ populate the full string despite any MaxCharacters parameter and not look to truncate it `OnLoad`. Given it reads the JSON, it feels like this could be the right behaviour but also feels inconsistent. Example is the numeric field will still be `3.1415926` despite testing with MaxCharacters = 4

**Comment by mdavisprog** [comment]
I agree about the inconsistency of loading a string from Json and it not respecting the MaxCharacters property. I went ahead and modified the the loading process to set the string for the TextInput control and this process will now truncate if MaxCharacters is set.

Thanks for the feature! It is very much appreciated.

**Pull request closed without merging**